spec/controllers/idv/personal_key_controller_spec.rb

```diff
@@ -52,81 +54,175 @@ def assert_personal_key_generated_for_profiles(*profile_pii_pairs)
       idv_session.address_verification_mechanism = 'phone'
       idv_session.user_phone_confirmation = true
       idv_session.vendor_phone_confirmation = true
+      idv_session.gpo_code_verified = false
     when 'gpo'
       idv_session.address_verification_mechanism = 'gpo'
       idv_session.user_phone_confirmation = false
       idv_session.vendor_phone_confirmation = false
+      idv_session.gpo_code_verified = true
+    when nil
+      idv_session.address_verification_mechanism = nil
+      idv_session.user_phone_confirmation = nil
+      idv_session.vendor_phone_confirmation = nil
     else
       raise 'invalid address_verification_mechanism'
     end
 
+    idv_session.threatmetrix_review_status = threatmetrix_review_status
+
     if mint_profile_from_idv_session
       idv_session.create_profile_from_applicant_with_password(password)
     end
   end
 
+  describe '#step_info' do
+    let(:step_info) do
+      controller.class.step_info
+    end
+
+    describe '#undo' do
+      it 'clears personal_key_acknowledged' do
+        idv_session.acknowledge_personal_key!
+        step_info.undo_step.call(idv_session: idv_session, user: user)
+        expect(idv_session.personal_key_acknowledged).to eql(nil)
+      end
+    end
+
+    describe '#preconditions' do
```

Comment:
I'm not sure whether it makes sense to test preconditions directly, or if it makes more sense to check for the behavior of rendering show or redirecting.

Reply:
I'm going with preconditions + an assertion that the confirm_step_allowed before_action is in place